File: packages/demo/test/models/post.test.js

import { beforeEach, test } from 'node:test';
import assert from 'node:assert';

import { Workspace, reset } from './helpers.js';

beforeEach(reset);

test(`post`, () => Workspace.run(async _ => {
    const { users, posts } = _.database;

    assert.equal(await posts.count(), 0);

    const user = await users.insert({
        name: 'Admin',
        email: 'admin@example.com',
        role: 'admin'
    });

    const { id } = await posts.insert({
        userId: user.id,
        title: 'Foo'
    });

    assert.equal(await posts.count(), 1);

    let post = await posts.where({ id }).first();

    assert.equal(post.title, 'Foo');


    assert.equal(await post.user.name, 'Admin');

    await post.update({ title: 'Bar' });

    post = await posts.where({ id }).first();

    assert.equal(post.title, 'Bar');

    await post.delete();

    assert.equal(await posts.count(), 0);
}));

File: projects/demo/test/models/post_test.rb

require "test_helper"

Class.new Minitest::Test do

    include Slick::Helpers

    def setup
        run_command('reset-database')
    end

    def test_post
        assert_equal 0, posts.count

        user = users.insert name: 'Admin', email: 'admin@example.com', role: 'admin'

        id = posts.insert(user_id: user.id, title: 'Foo').id

        assert_equal 1, posts.count

        post = posts.id_eq(id).first

        assert_equal 'Foo', post.title

        assert_equal 'Admin', post.user.name

        post.update title: 'Bar'

        post = posts.id_eq(id).first

        assert_equal 'Bar', post.title

        post.delete

        assert_equal 0, posts.count
    end

end
